src/libostree/ostree-repo-commit.c

```diff
@@ -819,7 +945,7 @@ write_object (OstreeRepo         *self,
 
   g_assert (actual_checksum != NULL); /* Pacify static analysis */
           
-  if (indexable && temp_file_is_regular)
```

Comment:
Might want to use upstream's variable here (`indexable` instead of `self->generate_sizes`)